maestro-android/src/androidTest/java/dev/mobile/maestro/MaestroDriverService.kt

```diff
@@ -108,6 +110,35 @@ class Service(
     private var locationCounter = 0
     private val toastAccessibilityListener = ToastAccessibilityListener.start(uiAutomation)
 
+    override fun launchApp(request: MaestroAndroid.LaunchAppRequest,
+                           responseObserver: StreamObserver<MaestroAndroid.LaunchAppResponse>) {
+        val context = InstrumentationRegistry.getInstrumentation().targetContext
+
+        val intent = context.packageManager.getLaunchIntentForPackage(request.packageName)
+
+        if (intent == null) {
+            Log.e("Maestro", "No launcher intent found for package ${request.packageName}")
+            responseObserver.onError(RuntimeException("No launcher intent found for package ${request.packageName}"))
+            return
+        }
+
+        request.argumentsList
+            .forEach { arg ->
+                val keyValue = arg.split("=")
+
+                if (keyValue.size == 2) {
+                    intent.putExtra(keyValue[0], keyValue[1])
+                } else {
+                    intent.putExtra(arg, true)
+                }
+            }
+
+        context.startActivity(intent)
```

Comment:
Did we also checked with apps that have multiple launchers? For example the ones with leak canary.

Reply:
Hm, isn't `getLaunchIntentForPackage` solving that already? I am not 100% certain

Reply:
I'm not sure too, let's verify maybe verify from the sample app we use.

Reply:
Verified should be handling ✅